src/OrchardCore/Orchard.Environment.Shell/ShellSettingsConfigurationProvider.cs

```diff
@@ -50,23 +50,14 @@ public void SaveToSource(string name, IDictionary<string, string> configuration)
                 Optional = false
             });
 
-            configurationProvider.Set(name, null);
-            configurationProvider.Set($"{name}:RequestUrlHost", ObtainValue(configuration,$"{name}:RequestUrlHost"));
-            configurationProvider.Set($"{name}:RequestUrlPrefix", ObtainValue(configuration,$"{name}:RequestUrlPrefix"));
-            configurationProvider.Set($"{name}:DatabaseProvider", ObtainValue(configuration,$"{name}:DatabaseProvider"));
-            configurationProvider.Set($"{name}:TablePrefix", ObtainValue(configuration,$"{name}:TablePrefix"));
-            configurationProvider.Set($"{name}:ConnectionString", ObtainValue(configuration,$"{name}:ConnectionString"));
-            configurationProvider.Set($"{name}:State", ObtainValue(configuration,$"{name}:State"));
+            foreach (var item in configuration)
+            {
+                configurationProvider.Set($"{item.Key}", item.Value ?? "~");
```

Comment:
Well the original goal was to store anything in this single provider.. that works great, until you have a mix of mutable and immutable state. Keys from a KMS shouldn't be stored locally for example. 

Maybe there's a better way?

I'm my solution at work I have two providers, one that brings in the keys from the KMS and one for local.